pygfx/resources/_buffer.py

```diff
@@ -135,15 +144,32 @@ def format(self):
 
     @property
     def vertex_byte_range(self):
-        """The offset and size, in bytes, when used as a vertex buffer."""
-        return self._vertex_byte_range
+        raise DeprecationWarning(
+            "vertex_byte_range is deprecated, use draw_range instead."
+        )
 
     @vertex_byte_range.setter
     def vertex_byte_range(self, offset_nbytes):
-        offset, nbytes = int(offset_nbytes[0]), int(offset_nbytes[1])
-        assert offset >= 0
-        assert offset + nbytes <= self.nbytes
-        self._vertex_byte_range = offset, nbytes
+        raise DeprecationWarning(
+            "vertex_byte_range is deprecated, use draw_range instead."
+        )
+
+    @property
+    def draw_range(self):
+        """The range to data (origin, size) expressed in items."""
+        return self._store.draw_range
+
+    @draw_range.setter
+    def draw_range(self, draw_range):
+        origin, size = draw_range
```

Comment:
* Why not make the signature `origin, size`?
* Why do values need to be multiplied by itemsize? Numbers should be expressed as the amount of indices to render for indexed geometry, or the amount of vertices for non-indexed geometry.

```suggestion
    def draw_range(self, origin, size):
```